src/effects/VST3/internal/ComponentHandler.h

```diff
@@ -0,0 +1,73 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file ComponentHandler.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#pragma once
+
+#include <memory>
+#include <atomic>
+#include <functional>
+#include <pluginterfaces/vst/ivsteditcontroller.h>
+
+namespace Steinberg
+{
+   namespace Vst
+   {
+      class IParameterChanges;
+   }
+}
+
+namespace internal
+{
+   /** Stores all pending changes from the editor, and
+     * safely transfers them to the processing thread on request.
+     * It's lock and wait free (given that there are two threads),
+     * but it's theoretically possible that
+     * processing thread will never get pending changes, but in real life
+     * should never be the case.
+     * Current implementation isn't strict about begin/perform/endEdit
+     * order correctness(actually it's completely ignored right now,
+     * but that might change in future).
+     */
+   class ComponentHandler : public Steinberg::Vst::IComponentHandler
+   {
+      Steinberg::int32 mSampleOffset { 0 };
+
+      std::atomic<Steinberg::Vst::IParameterChanges*> mFirst { nullptr };
+      std::atomic<Steinberg::Vst::IParameterChanges*> mSecond { nullptr };
+      std::atomic<Steinberg::Vst::IParameterChanges*> mPendingChanges { nullptr };
+
+   public:
+      using PendingChangesPtr = std::unique_ptr<
+         Steinberg::Vst::IParameterChanges,
+         std::function<void (Steinberg::Vst::IParameterChanges*)>>;
+
+      ComponentHandler();
+      virtual ~ComponentHandler();
+
+      //Sets the current sample offset, which will be assigned
+      //to the points created by subsequent calls to performEdit 
+      void setSampleOffset(Steinberg::int32 offset) noexcept;
```

Comment:
I see that this function is new since my last round of review but not used.  (It was declared then, but not defined.  Now it is defined but never called.)

Reply:
I'll remove it, we don't need it now, since we don't have automation support yet